homeassistant/components/modbus/light.py

```diff
@@ -0,0 +1,41 @@
+"""Support for Modbus lights."""
+import logging
+from typing import Optional
+
+from homeassistant.components.light import LightEntity
+from homeassistant.const import CONF_LIGHTS, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import MODBUS_DOMAIN
+from .modbus import ModbusHub
+from .switch import ModbusSwitch
```

Comment:
A platform is not allowed to depend on another platform. Please refactor this.

Reply:
may I ask what is the reason for this ? it duplicates code. Is it legal if we (again) make a switchBase, and switch inherit it with an empty class?

Reply:
As @janiversen mentioned, we're trying to avoid code duplication. Do you have any suggestions what would be acceptable for you?

Reply:
Make a class outside a platform that only inherits `Entity` without platform specific methods. The platform specific methods, eg `turn_on` should only be implemented in each platform. You could add a helper method in the common class for the common work that needs to be done in `turn_on`.

Reply:
got it. Gracias.

@vzahradnik I recommend you duplicate the switch code for now. Then we can later make a base class.

Reply:
OK, I will make the changes.